rotkehlchen/exchanges/coinbase.py

```diff
@@ -160,20 +172,113 @@ def __init__(
             secret=secret,
             database=database,
         )
+        try:
+            self.is_legacy_api_key = self.is_legacy_key(api_key)
+        except AuthenticationError as e:
+            self.is_legacy_api_key = True
+            log.error(f'Error determining API key format: {e}. Defaulting to legacy key.')
+
+        # Set CB-ACCESS-KEY header if its a legacy key format
+        if self.is_legacy_api_key:
+            self.session.headers.update({'CB-ACCESS-KEY': self.api_key})
+            self.session.headers.update({'CB-VERSION': CB_VERSION})
```

Comment:
Shoudl be combined. Doing it

```suggestion
            self.session.headers.update({'CB-ACCESS-KEY': self.api_key, 'CB-VERSION': CB_VERSION})
```